src/ynViewer.sol

```diff
@@ -23,7 +26,40 @@ contract ynViewer is IynViewer {
     }
 
     /// @inheritdoc IynViewer
-    function getAllStakingNodes() public view returns (IStakingNode[] memory) {
-        return stakingNodesManager.getAllNodes();
+    function getRate() external view returns (uint256) {
+        uint256 _totalSupply = ynETH.totalSupply();
+        uint256 _totalAssets = ynETH.totalAssets();
+        if (_totalSupply == 0 || _totalAssets == 0) return 0;
+        return 1 ether * _totalAssets / _totalSupply;
+    }
+
+    /// @inheritdoc IynViewer
+    function withdrawalDelayBlocks(address _strategy) external view returns (uint256) {
+        IDelegationManager _delegationManager = stakingNodesManager.delegationManager();
+        uint256 _minDelay = _delegationManager.minWithdrawalDelayBlocks();
+        uint256 _strategyDelay = _delegationManager.strategyWithdrawalDelayBlocks(IStrategy(_strategy));
+        return _minDelay > _strategyDelay ? _minDelay : _strategyDelay;
+    }
+
+    /// @inheritdoc IynViewer
+    function getStakingNodeData() external view returns (StakingNodeData[] memory _data) {
+        IStakingNode[] memory _nodes = stakingNodesManager.getAllNodes();
+
+        uint256 _length = _nodes.length;
+        _data = new StakingNodeData[](_length);
+
+        for (uint256 i = 0; i < _length; ++i) {
+            IStakingNode _node = _nodes[i];
+            IEigenPod _eigenPod = IEigenPod(stakingNodesManager.eigenPodManager().getPod(address(_node)));
+            _data[i] = StakingNodeData({
+                nodeId: _node.nodeId(),
+                ethBalance: _node.getETHBalance(),
+                eigenPodEthBalance: _eigenPod.nonBeaconChainETHBalanceWei(),
```

Comment:
the meaning of this was intended to be:

address(_eigenPod).balance

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/119/commits/c2c704acee8d56577dcd3f571e7e671a3140813d